### DISCUSSION TITLE
HERE API help needed

### DISCUSSION THREAD

**Discussion by hellkaim** [rinigus/pure-maps#656] [Q&A]
As you may all know there are few providers with traffic available.
One of them is HERE.

As requested I created my own developer account and am trying to add an API key. My platform is UBPorts and the issue is that the API key field has a limit in terms of length.

So as there are several endpoints (REST and OAuth) I can't get exactly what should I enter in the application. There also is a API reference that... is what?

In each case all is working except traffic. The screen gets gray and that is it. So any advice would be appreciated.

**Comment by jonnius** [comment]
About the textfield: Do you mean a limit of what it displays or the number of characters it accepts? I can't reproduce the latter one. I can paste my key there and everything works including traffic info.

About which key: It seems they are restructuring their website currently. I do find my key navigating to "Projects" via the top right menu button. Create a freemium one and create a JavaScript API key.

**Comment by hellkaim** [comment]
@jonnius Thank you for the reply.
As I am using a mobile device there is no way to correctly check is it a " limit of what it displays or the number of characters it accepts". When I pasted the API key for HERE navigate to the main menu (press the button on the top left corner of the screen) and then relaunch the app - I can see, select and copy only about 20 characters.

In any case despite the entered amount the layers and maps seems not to work.

So. What am I doing:

1. Go to [platform.here.com](platform.here.com) and authenticate myself
2. From the right panel selecting Access Manager
3. Select a user name
4. Select an app
5. In Credentials switching to REST API keys
6. Create or Copy depending on if the key was created during the APP setup.

Now having that key I go to my device and enter it in the HERE API string in the UI. Restart the app... and see a gray screen instead of traffic layer.

**Reply by jonnius** [reply]
About the textfield: I can add more than enough characters to the textfield. For navigating the cursor better, use the virtual touch pad you can access with a swipe from the bottom edge or by holding the space key.

About the key: Could you please try the JavaScript API key as mentioned above?

**Comment by rinigus** [comment]
I have been using https://developer.here.com , not platform. Try with that

**Comment by hellkaim** [comment]
Sory for delay.
HERE does not provide any way to registsr a new login anymore. They are redirecting to platform.here.com

What should be my actions?

**Reply by jonnius** [reply]
I found my key on platform.here.com, too. It is like you describe above, except that under **Credentials**  I see the sub sections **OAuth 2.0** and **API Keys**. My key is under the latter one. I don't see any **REST API Keys** section, which might be the wrong section anyways.

**Reply by hellkaim** [reply]
I tryed that. My map dose not load. The only thing working -sattelite images. And there seems ro be no permissions or alike.
I am just trying to understand what and where should i enable. There is no console or debug that I am aware of to check what's wrong (

**Reply by hellkaim** [reply]
Ok. I entered it again and it seems to work except for traffic. Should I anable it somewhere?
With "default" key it seems to work.

**Reply by hellkaim** [reply]
Edited the projct - all services are added. Still no traffic layer (((

**Reply by jonnius** [reply]
All I can say is that in the project view, no service is linked in my case, but I see transit info as one of the services I could link. Might still be incomplete for me due to their ongoing migrations.

**Comment by hellkaim** [comment]
Well, I added the " API Keys" key and it seems to work. Still without traffic. If you will able to find the answer (or any one) - please update the issu.

**Reply by rinigus** [reply]
did you try to run it in terminal and see standard output? I would expect some error messages if the key was not used. How did you check for traffic? Did you select map traffic mode (top right button)?

**Reply by hellkaim** [reply]
Not in terminal, this is a good idea!

The traffic - yes, i am switching layers with the help of the button in the upper-right corner of the interface.
